babel/localtime/_helpers.py

```diff
@@ -0,0 +1,43 @@
+try:
+    import zoneinfo
+except ModuleNotFoundError:
+    zoneinfo = None
+    import pytz
+
+
+def _get_tzinfo(tzenv):
+    """Get the tzinfo from `zoneinfo` or `pytz`
+
+    :param tzenv: timezone in the form of Continent/City
+    :return: tzinfo object or None if not found
+    """
+    if zoneinfo:
+        try:
+            return zoneinfo.ZoneInfo(tzenv)
+        except zoneinfo.ZoneInfoNotFoundError:
+            pass
+
+    else:
+        try:
+            return pytz.timezone(tzenv)
+        except pytz.UnknownTimeZoneError:
+            pass
+
+    return None
+
+def _get_tzinfo_or_raise(tzenv):
+    tzinfo = _get_tzinfo(tzenv)
+    if tzinfo is None:
+        raise LookupError(
+            f"Can not find timezone {tzenv}. \n"
+            "Please use a timezone in the form of Continent/City"
```

Comment:
Maybe something like
```suggestion
            f"Can not find timezone {tzenv}. "
            "Timezone names are generally in the form `Continent/City`."
```
because there are `Etc/` timezones that aren't continent-y.